### PR TITLE
Jetpack connect: Use makeJsonSchemaParser

### PR DESCRIPTION
**Pull request by sirreal**
A parser was being used without using the parser lib. After #24288 ✅, we'll be able to leverage makeJsonSchemaParser, which is much nicer.

This PR also abstracts some duplicated logic into a function what wraps the parser, as well as lazy initialization for the parser.

No functional changes.

## Testing
1. Go through an authorization flow logged in and logged out. Technically a logged out flow should cover everything.
1. Everything should behave normally.
1. Ensure that you seen the warning here https://calypso.live/jetpack/connect/authorize?branch=update/jpc-make-parser

### CODE REVIEW

File: client/jetpack-connect/controller.js [hunk 1]
```diff
@@ -204,19 +228,17 @@ export function signupForm( context, next ) {
...
-		let interval = context.params.interval;
-		let locale = context.params.locale;
-		if ( context.params.localeOrInterval ) {
-			if ( [ 'monthly', 'yearly' ].indexOf( context.params.localeOrInterval ) >= 0 ) {
-				interval = context.params.localeOrInterval;
+		let { interval, locale } = context.params;
+		const { localeOrInterval } = context.params;
+		if ( localeOrInterval ) {
+			if ( includes( [ 'monthly', 'yearly' ], localeOrInterval ) ) {
+				interval = localeOrInterval;
```
Comment (sirreal) [inline client/jetpack-connect/controller.js]: I'll roll back the changes in this section. They're out of scope, and interval appears to be completely irrelevant for signup/authorize. I'll handle that in another PR.

File: client/jetpack-connect/controller.js [hunk 2]
```diff
@@ -88,6 +88,30 @@ const getPlanSlugFromFlowType = ( type, interval = 'yearly' ) => {
 	return get( planSlugs, [ interval, type ], '' );
 };
 
+/**
+ * !! Private !!
+ *
+ * Lazy initialized authorization query parser.
+ */
+let _authQueryParser = null;
```
Comment (tyxla) [inline client/jetpack-connect/controller.js]: If we want to keep this private, we could move it inside `parseAuthorizationQuery` as a static variable. Could also help remove the leading underscore, and the need to have an initial ` = null` declaration.

**Comment by sirreal** [comment]
Needs a rebase after #24314.

Nice thought @tyxla, I'll address that as well

**Comment by sirreal** [comment]
Added tests, rebased, applied suggestion and reorganized a bit (moved new function into utils).

This could use a smoke test.

File: client/jetpack-connect/test/utils.js [hunk 3]
```diff
@@ -60,3 +65,27 @@ describe( 'getRoleFromScope', () => {
...
+			home_url: 'http://yourjetpack.blog',
+			redirect_uri: 'http://yourjetpack.blog/wp-admin/admin.php',
+			scope: 'administrator:34579bf2a3185a47d1b31aab30125d',
+			secret: '640fdbd69f96a8ca9e61',
+			site: 'http://yourjetpack.blog',
+			site_url: 'http://yourjetpack.blog',
+			state: '1',
+		};
+		const result = parseAuthorizationQuery( data );
+		expect( result ).not.toBeNull();
```
Comment (tyxla) [inline client/jetpack-connect/test/utils.js]: This assertion seems unnecessary IMHO, considering the snapshot contains an object.

Comment (sirreal) [reply client/jetpack-connect/test/utils.js]: The snapshot _could be_ of null. Snapshot tests are, in some ways, opaque. Especially when you first add them. I believe there's value in documenting explicit expectations in addition to snapshots.

**Pull request merged**